### PR TITLE
Jane's API-Muncher

### PR DESCRIPTION
**Pull request by JaneEdwMcN**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
How did you go about exploring the Edamam API, and how did you try querying the API? | I read through the Edamam Recipe API documentation and practiced querying it using Postman.
What is an API Wrapper? Why is it in `lib`? How would your project change if you needed to interact with more than one API (aka more than just the Edamam API)? | An API Wrapper is a file that separates concerns between the Rails app and the API calls. It's in the lib folder because it is a custom library. If I needed to interact with more than one API I would create another API Wrapper for that API to keep the different APIs separate.
Describe your API wrapper, the methods you created in it, and why you decided to create those methods/how they were helpful | My API wrapper is in my lib folder and has methods that query the Edamam API. I have a create recipe method that takes the data received from the API and creates an instance of Recipe. I have a list_recipes method that takes a words as a parameter and searches the Edamam API with those words. If there are matches with Edamam, it takes those matches and creates recipes with them. This method is called in the Recipes Controller and enables recipes to be displayed in the index view page. I have a find_recipe method that takes the unique part of a Edamam recipe’s URI and searches Edamam’s API for a specific recipe. If a recipe is found the data is used to create a single instance of the Recipe class. This is called in the show method in the Recipes Controller and then displayed on the show view page.
What was an edge case or failure case test you wrote for your API Wrapper? What was a nominal case? | An edge case would be if the user attempted to search for recipes without typing anything or with a word that doesn’t exist, they will be redirected and a flash message will display. A nominal case would be that create_recipe creates a recipe with valid parameters.
How does VCR aid in testing an API? | VCR makes tests run faster because you don’t have to query the API every time the tests are run. VCR “records” API queries and replays them instead of making a new query. This also saves money or consequences for a site that has a finite amount of queries allowed to the API they are using.
What is the Heroku URL of your deployed application? | http://no-nana.herokuapp.com

### CODE REVIEW

File: app/views/recipes/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<section>
+  <h2 class ="index-heading">Recipes with "<%=@ingredients%>"</h2>
```
Comment (CheezItMan) [inline app/views/recipes/index.html.erb:2]: unclosed `h2` element

File: lib/edamam_api_wrapper.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      data["hits"].each do |recipe_data|
+        recipe_list << create_recipe(recipe_data["recipe"])
+      end
+      return recipe_list
+    end
+
+  end
+
+  def self.find_recipe(id)
+    url = BASE_URL + "search?" + "r=http%3A%2F%2Fwww.edamam.com%2Fontologies%2Fedamam.owl%23recipe_#{id}" + "&app_id=#{ID}&app_key=#{KEY}"
```
Comment (CheezItMan) [inline lib/edamam_api_wrapper.rb:24]: Minor style thing, you have one variable called `id` and another called `ID`, could be slightly confusing.

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Core Requirements** |
Git hygiene |  Good commits and good commit messages
Comprehension questions |  Check, 
**General** |
Rails fundamentals (RESTful routing, use of named paths) |  Check
Semantic HTML |  Pretty semantic html
Errors are reported to the user | Check, good use of `flash`
API Wrapper to handle the API requests |  Check
Controller testing |  Check, very well done.  
Lib testing |  I would have made the `create_recipe` method a private helper method and not tested it except through the search and find methods.  
Search Functionality |  Check
List Functionality | Check
Show individual item functionality |  Check
**Styling** |
List view shows 10 items at a time and/or has pagination |  Check
Attractive and usable UI |  Check, very nice!  Good use of cards from Bootstrap
**API Features** |
The App attributes Edamam |  Check
The VCR cassettes do not contain the API key |  Check
**External Resources** |
Link to deployed app on Heroku |  Check
**Overall** |  Awesome work, you hit all the learning goals and did it well.  Nice work!
